pkg/kubectl/cmd/drain.go

```diff
@@ -227,28 +234,65 @@ func (o *DrainOptions) getPodsForDeletion() ([]api.Pod, error) {
 				// gone/missing, not for any other cause.  TODO(mml): something more
 				// sophisticated than this
 				if err == nil && ds != nil {
-					replicated = true
+					// Otherwise, treat daemonset-managed pods as unmanaged since
+					// NodeController currently ignores the unschedulable bit.
+					// FIXME(mml): Add link to the issue concerning a proper way to drain
+					// daemonset pods, probably using taints.
+					daemonset_pod = true
 				}
 			}
 		}
-		if replicated || o.Force {
-			pods = append(pods, pod)
-		}
-		if !replicated {
+		switch {
+		case daemonset_pod:
+			daemonSetPodNames = append(daemonSetPodNames, pod.Name)
+		case !replicated:
 			unreplicatedPodNames = append(unreplicatedPodNames, pod.Name)
+			if o.Force {
+				pods = append(pods, pod)
+			}
+		default:
+			pods = append(pods, pod)
 		}
 	}
 
-	if len(unreplicatedPodNames) > 0 {
-		joined := strings.Join(unreplicatedPodNames, ", ")
-		if !o.Force {
-			return pods, fmt.Errorf("refusing to continue due to pods managed by neither a ReplicationController nor a DaemonSet: %s (use --force to override)", joined)
+	if (!o.IgnoreDaemonsets && len(daemonSetPodNames) > 0) || (!o.Force && len(unreplicatedPodNames) > 0) {
+		if o.Force {
+			return pods, errors.New(unmanagedMsg([]string{}, daemonSetPodNames, true))
+		} else {
+			return pods, errors.New(unmanagedMsg(unreplicatedPodNames, daemonSetPodNames, true))
+		}
+	} else if len(unreplicatedPodNames) > 0 {
+		// If we get here, we know o.Force is true.
+		fmt.Fprintf(o.out, "WARNING: About to delete these %s\n", unmanagedMsg(unreplicatedPodNames, []string{}, false))
+		if len(daemonSetPodNames) > 0 {
+			fmt.Fprintf(o.out, "WARNING: Skipping %s\n", unmanagedMsg([]string{}, daemonSetPodNames, false))
```

Comment:
I think we also want to print this message in the case where unreplicatedPodNames == 0 && o.IgnoreDaemonsets && len(daemonSetPodNames) > 0, but if I'm understanding the logic correctly, you won't do that.